.github/workflows/publish-iceberg-rest-fixture-docker.yml

```diff
@@ -20,9 +20,8 @@
 name: Build and Push 'iceberg-rest-fixture' Docker Image
 
 on:
-  push:
-    tags:
-      - 'apache-iceberg-[0-9]+.[0-9]+.[0-9]+'
```

Comment:
let's keep both, we still need the tag for 
https://github.com/apache/iceberg/pull/12008/files#diff-24cba6867a8ec1ac782a5dbfb5ec71f84ed7377564afeada5661647f2b480879L47-L51

This way we have a tag for each release and another for `latest`

Reply:
we dont have any tag releases yet since this code was added after the last tag https://github.com/apache/iceberg/tags

Reply:
Ah, good catch!